.github/workflows/java-roll-version-and-release.yaml

```diff
@@ -0,0 +1,110 @@
+# A workflow to find the current version from pom and then commit the release version (with tag) and the next dev cycle version
+name: Roll maven version and release
+on:
+  workflow_call:
+    inputs:
+      # TODO remove these 2 params once we are out of OpenStack and build-locker build is not needed anymore
+      distribution:
+        description: 'buildlocker distribution name'
+        type: string
+        required: true
+      type:
+        description: 'buildlocker distribution type [war|jar*]. Use any other value to skip pushing to build-locker'
+        type: string
+        required: false
+
+
+jobs:
+  extract:
+    name: extract-version
+    runs-on:
+      - self-hosted
+      - java
+    outputs:
+      major: ${{ steps.major.outputs.major }}
+      minor: ${{ steps.minor.outputs.minor }}
+      patch: ${{ steps.patch.outputs.patch }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: major
+        run: |
+          echo "major=$(mvn --quiet build-helper:parse-version help:evaluate -Dexpression=parsedVersion.majorVersion -DforceStdout --no-transfer-progress)" >> $GITHUB_OUTPUT
+      - id: minor
+        run: |
+          echo "minor=$(mvn --quiet build-helper:parse-version help:evaluate -Dexpression=parsedVersion.minorVersion -DforceStdout --no-transfer-progress)" >> $GITHUB_OUTPUT
+      - id: patch
+        run: |
+          echo "patch=$(mvn --quiet build-helper:parse-version help:evaluate -Dexpression=parsedVersion.incrementalVersion -DforceStdout --no-transfer-progress)" >> $GITHUB_OUTPUT
+      - id: echo-version
+        run: |
+          echo ${{steps.major.outputs.major}}.${{steps.minor.outputs.minor}}.${{steps.patch.outputs.patch}}
+  update-version-commit:
+    needs: extract
+    env:
+      NEW_VERSION: ${{ needs.extract.outputs.major }}.${{ needs.extract.outputs.minor }}.${{ needs.extract.outputs.patch }}
+    runs-on:
+      - self-hosted
+      - java
+    steps:
+      - run: echo "NEW_VERSION=$NEW_VERSION"
+      - uses: actions/checkout@v4
+      - name: Setup git author information
+        run: |
+          git config user.name RCSBSystem
```

Comment:
It might not be important, but GitHub's docs recommend this:

```bash
git config --global user.name 'github-actions[bot]'
git config --global user.email 'github-actions[bot]@users.noreply.github.com'
```